### PR TITLE
Add github actions workflows upon pull request and push actions

### PR DESCRIPTION
**Pull request by shawngerrard**
Rationale and approach as per PR [shawngerrard/asterion-as-code#2](https://github.com/shawngerrard/asterion-as-code/pull/2#issue-1199350213).

### CODE REVIEW

File: .github/workflows/asterion-aws-pr.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: asterion-aws-infra-ci-pr
```
Comment (jmhbnz) [inline .github/workflows/asterion-aws-pr.yml]: Minor but would prefer name follows directory, i.e. `asterion-infra-aws-ci-pr`.

File: .github/workflows/asterion-aws-push.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: asterion-aws-infra-ci-push
```
Comment (jmhbnz) [inline .github/workflows/asterion-aws-push.yml]: Minor but would prefer name follows directory, i.e. `asterion-infra-aws-ci-push`.

File: .github/workflows/asterion-aws-pr.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Import Github actions
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (jmhbnz) [inline .github/workflows/asterion-aws-pr.yml]: We'll need to populate these in github: https://github.com/asterion-digital/asterion-as-code/settings/secrets/actions

Do we have an organisation level account set up in aws for asterion?

File: .github/workflows/asterion-aws-pr.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          aws-region: ${{ secrets.AWS_REGION }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      - name: Install Pulumi project dependencies
+        run: pip install -r requirements.txt
+      - name: Select Pulumi stack and configure project
+        run: pulumi stack select dev && echo "$AWS_PUBLIC_KEY" | pulumi config set publickey && echo "$AWS_PRIVATE_KEY" | pulumi config set privatekey && echo "ap-southeast-2" | pulumi config set aws:region
```
Comment (jmhbnz) [inline .github/workflows/asterion-aws-pr.yml]: Can we make this `run:` command multiline so it's easier to read?

**Review by jmhbnz — CHANGES_REQUESTED** [review]
Nice work getting this underway @shawngerrard.   Just a few things we need to work through, let me know what you think 👍🏻

File: .github/workflows/asterion-aws-pr.yml (hunk 3 above)
Comment (shawngerrard) [reply .github/workflows/asterion-aws-pr.yml]: Not yet - I'll need to create the organization and update users, account billing, etc to point to Asterion assets.

**Comment by shawngerrard** [comment]
Thanks @jmhbnz, good feedback. I'll set about implementing these changes as well as updates to the guide.

**Pull request merged**